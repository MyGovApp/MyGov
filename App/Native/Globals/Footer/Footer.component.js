import React from 'react'
import { View } from 'react-native'
import s from './Footer.styles.js'
import { images as I } from '../../Themes'
import IconButton from '../IconButton'

const Footer = () => {
  const footerItems = [
    { route: 'bills', text: 'Bills', image: I.billsIcon },
    { route: 'congress', text: 'Congress', image: I.congressIcon },
    { route: 'notifications', text: 'Notifications', image: I.notificationIcon },
    { route: 'dev', text: 'Dev', image: I.devIcon }
  ]

  return (
    <View style={s.footerContainer} >
      {footerItems.map((item, i) => <IconButton key={i} {...item} />)}
    </View>
  )
}

module.exports = Footer
