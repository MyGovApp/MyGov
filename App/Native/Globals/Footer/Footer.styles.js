import { colors } from '../../Themes'

const styles = {
  footerContainer: {
    backgroundColor: colors.primaryBlue,
    flexDirection: 'row'
  }
}

export default styles
