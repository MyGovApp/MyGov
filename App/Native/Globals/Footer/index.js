import Footer from './Footer.component.js'

export default Footer
