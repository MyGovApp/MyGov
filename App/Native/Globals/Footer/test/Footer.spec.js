import 'react-native'
import React from 'react' // eslint-disable-line
import renderer from 'react-test-renderer'
import { ComponentWrapper } from './Footer.setup'

describe('Native Footer', () => {
  it('should render the Footer component correctly', () => {
    const tree = renderer.create(ComponentWrapper()).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
