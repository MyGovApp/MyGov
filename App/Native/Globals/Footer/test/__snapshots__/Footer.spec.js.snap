// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Native Footer should render the Footer component correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#112E50",
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "flex": 1,
          "height": 50,
          "paddingBottom": 5,
          "paddingTop": 5,
        },
      ]
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "padding": 5,
            "width": 25,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "textAlign": "center",
          }
        }
      >
        Bills
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "flex": 1,
          "height": 50,
          "paddingBottom": 5,
          "paddingTop": 5,
        },
      ]
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "padding": 5,
            "width": 25,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "textAlign": "center",
          }
        }
      >
        Congress
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "flex": 1,
          "height": 50,
          "paddingBottom": 5,
          "paddingTop": 5,
        },
      ]
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "padding": 5,
            "width": 25,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "textAlign": "center",
          }
        }
      >
        Notifications
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "flex": 1,
          "height": 50,
          "paddingBottom": 5,
          "paddingTop": 5,
        },
      ]
    }
    testID={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "padding": 5,
            "width": 25,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "textAlign": "center",
          }
        }
      >
        Dev
      </Text>
    </View>
  </View>
</View>
`;
